### PR TITLE
#18 - Refactoring and Docker

### PR DESCRIPTION
**Pull request by denis-onder**
Issue #18 

TL;DR - What does this implement?
---------------------------------
Satisfies the requirements of #18, added a Docker configuration file and did some slight code tweaks.

How to Test
-----------
Run `docker-compose up`

Checklist
---------

 - [ ] Do we need to test this?
 - [ ] Were tests written?
 - [x] Does this meet coding standards?
 - [x] Is this is code change?
 - [ ] Is this a meta change (e.g issue template)?
 - [x] Is this pull request ready for review?
 - [ ] Is this a breaking change (e.g not backwards compatible with clients)?
 
ESLint Output
-------------

```
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.
```

Reviewers
---------

@gmemstr

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -13,7 +13,7 @@ MAIL_SENDER=
...
 GOOGLE_OAUTH_CLIENT_ID=
 GOOGLE_OAUTH_CLIENT_SECRET=
-TWITTER_OAUTH_CONSUMER_KEY=
-TWITTER_OAUTH_CONSUMER_SECRET=
 DISCORD_OAUTH_CLIENT_ID=
-DISCORD_OAUTH_CLIENT_SECRET=
\ No newline at end of file
+DISCORD_OAUTH_CLIENT_SECRET=
+TWITTER_OAUTH_CONSUMER_KEY=
+TWITTER_OAUTH_CONSUMER_SECRET=
```
[the remark this answers is not in the document]
Comment (denis-onder) [reply .env.example]: OCD :smile:

File: README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # NachoNight Authentication API
 
+> Documentation: [https://github.com/NachoNight/documentation/blob/master/docs/auth/README.md](https://github.com/NachoNight/documentation/blob/master/docs/auth/README.md)
```
[the remark this answers is not in the document]
Comment (denis-onder) [reply README.md]: Removed blockquote

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
   "homepage": "https://github.com/NachoNight/nacho-night-auth-api#readme",
   "dependencies": {
     "axios": "^0.19.0",
-    "bcrypt": "^3.0.6",
+    "bcryptjs": "^2.4.3",
```
[the remark this answers is not in the document]
Comment (denis-onder) [reply package.json]: The bcrypt package, for some odd reason, won't install successfully when `npm install` is ran when trying to create a Docker image. The bcryptjs package is the exact same thing as bcrypt, but it works with Docker.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+WORKDIR /app
+
+COPY ./package.json .
+
+COPY ./package-lock.json .
+
+RUN npm install
+
+ENV DB_URL=postgres://user:pass@postgres:5432/db
```
[the remark this answers is not in the document]
Comment (denis-onder) [reply Dockerfile]: Moved environmental variables to the `docker-compose.yml` file

File: app/db/index.js [hunk 5]
```diff
@@ -4,20 +4,26 @@ const { readFileSync } = require('fs');
...
+    },
+    options: {
+      host,
+      dialect,
+      port,
+      logging: false,
+    },
   };
-  if (server.environment === 'staging') {
+  if (server.environment === 'staging' || server.environment === 'production') {
```
[the remark this answers is not in the document]
Comment (denis-onder) [reply app/db/index.js]: Just realized this one, changed to `if (server.environment !== 'development')`

File: app/db/index.js [hunk 6]
```diff
@@ -4,20 +4,26 @@ const { readFileSync } = require('fs');
...
+    },
+    options: {
+      host,
+      dialect,
+      port,
+      logging: false,
+    },
   };
-  if (server.environment === 'staging') {
+  if (server.environment !== 'development') {
```
Comment (gmemstr) [inline app/db/index.js]: This assumes that people will set the env var to `development` -- should we assume this?

File: app/db/index.js [hunk 7]
```diff
@@ -4,20 +4,26 @@ const { readFileSync } = require('fs');
...
-  const { name, username, password, host, dialect, port } = database;
+  const { URL, name, username, password, host, dialect, port } = database;
   const config = {
-    host,
-    dialect,
-    port,
-    logging: false,
+    connection: () => {
+      if (URL) return URL;
+      return `postgres://${username}:${password}@localhost:5432/${name}`;
```
Comment (gmemstr) [inline app/db/index.js]: `localhost:5432` hardcoded? Shouldn't this come from the env config? Or am I misreading this.

Comment (denis-onder) [reply app/db/index.js]: I didn't even see this one :smile:

File: app/db/index.js (hunk 6 above)
Comment (denis-onder) [reply app/db/index.js]: Well, the `environment` object key is set to `development` by default in the configuration, unless another value has been set in the `.env` file.

Comment (denis-onder) [reply app/db/index.js]: I could set it to `if (server.environment === 'staging')`, but if we'd put this on a production server, and set the `NODE_ENV` environmental variable to anything else than staging, the SSL configuration which should be applied by that block would not work.

Comment (denis-onder) [reply app/db/index.js]: Set to `if (server.environment === 'staging')`

File: app/db/index.js [hunk 8]
```diff
@@ -8,16 +8,15 @@ const connect = () => {
...
     },
     options: {
       host,
       dialect,
-      port,
       logging: false,
     },
   };
-  if (server.environment !== 'development') {
+  if (server.environment === 'staging') {
```
Comment (gmemstr) [inline app/db/index.js]: We'll want this to be active in production as well.

Comment (denis-onder) [reply app/db/index.js]: Added this to the if statement
